include/ideep/operators/inner_product.hpp

```diff
@@ -562,6 +563,9 @@ struct inner_product_backward_data : public dnnl::inner_product_backward_data {
     if (diff_dst.get_data_type() == data_type::bf16) {
       weights_.init(weights.get_desc().to_type(data_type::bf16));
       weights_.reorder_from(weights);
+    } else if (diff_dst.get_data_type() == data_type::f16) {
```

Comment:
I think there has a performance issue, why do we always have a reorder, I think we can remove this reorder if we can make sure the weights' type is always aligned with dst, if we can't make sure it, we can check it and only do this reorder when weight has different dtype with dst.

Reply:
>we can remove this reorder if we can make sure the weights' type is always aligned with dst, if we can't make sure it, we can check it and only do this reorder when weight has different dtype with dst.

Yes, I also think we can check it and only do this reorder when weight has different dtype with dst. Actually I followed bf16, should we do the same for bf16 ?